controllers/root.js

```diff
@@ -39,16 +39,23 @@ var Root = (function() {
 
             }
 
-    		IDM.check_token(auth_token, function (user_info) {
-
+    		IDM.check_token(auth_token, function (user_info) {                
+                
                 if (config.azf.enabled) {
                     
                     AZF.check_permissions(auth_token, user_info, req, function () {
+                        if (config.work_mode && config.work_mode === 'validate'){
+                            redir_request_validate(req, res, user_info);
+                        } else if (config.work_mode &&  config.work_mode === 'proxy'){
+                            redir_request(req, res, user_info);
+                        }
+                    }, function (status, e) {
 
-                        redir_request(req, res, user_info);
+                        if (status === 401 && config.work_mode && config.work_mode === 'validate') {
+                            log.error('User access-token not authorized: ', e);
+                            res.status(401).send('0');
```

Comment:
I think it's better to follow the next pattern in the responses (instead of sending 1 or 0): 
- If the validation fails, send the fail status code (500, 401 or whatever) and a message with the explanation (e.g. "User token not authorized")
- If the validation success, send a 200 ok response